component/stream/stream.go

```diff
@@ -245,57 +176,74 @@ func (st *Stream) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle,
 		handle = handlemap.NewHandle(options.Name)
 	}
 	if !st.streamOnly {
-		fileKey := options.Name
-		if st.cacheOnFileHandle {
-			handle.ID = handlemap.HandleID(nextHandleID.Inc())
-			fileKey = strconv.FormatUint((uint64(handle.ID)), 10)
+		if st.cache.openHandles >= st.cache.handleLimit {
+			err = errors.New("handle limit exceeded")
+			log.Err("Stream::OpenFile : error %s [%s]", options.Name, err.Error())
+			return handle, err
 		}
-
-		flock := st.fileKeyLocks.Get(fileKey)
-		flock.Lock()
-		st.streamCache.addFileKey(fileKey)
-		block, exists, _ := st.fetchBlock(fileKey, handle, 0)
+		atomic.AddInt32(&st.cache.openHandles, 1)
+		cacheObj := handlemap.Cache{
+			DataBuffer: common.NewLRUCache(int64(st.cache.bufferSizePerHandle)),
+		}
+		handle.CacheObj = &cacheObj
```

Comment:
Shall we have a method in handlemap to do this where we can just call handle.CreateCacheObject and pass on the size ?

Reply:
just in case later we need to add more stuff related to policy or other things, it will be easier to have it as a seperate method.